base/broadcast.jl

```diff
@@ -10,6 +10,11 @@ export broadcast_getindex, broadcast_setindex!
 
 ## Broadcasting utilities ##
 
+# fallback routines for broadcasting with no arguments or with scalars
+# to just produce a scalar result:
+broadcast(f) = f()
+broadcast(f, x::Number...) = f(x...)
```

Comment:
is the deprecation method already more specific? integer vs number? or is the union making dispatch pick Number?